### ISSUE TITLE
Question on git tags and versioning

### ISSUE THREAD

**Issue by jmstover** [open-mpi/ompi#7400]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
3.1.5 ... build from v3.1.5 tag

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
git clone

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.
```
# git submodule status
#
```

### Please describe the system on which you are running

* Operating system/version:  CentOS 7
* Computer hardware:  Dell R540
* Network type: TCP

-----------------------------

## Details of the problem

I checked out the v3.1.5 tag. But the build is the rc1 tag. The VERSION file includes: `greek=rc1`

```
$ mpiexec --version
mpiexec (OpenRTE) 3.1.5rc1

Report bugs to http://www.open-mpi.org/community/help/
```

My question is if this is expected from a git clone. Or if the 3.1.5 release is supposed to be marked as rc1 (it's the same with 4.0.2). The release 3.1.5 release tarball also sets the version to rc1.

**Note**: If you include verbatim output (or a code block), please use a [GitHub Markdown](https://help.github.com/articles/creating-and-highlighting-code-blocks/) code block like below:
```shell
shell$ mpirun -np 2 ./hello_world
```

**Comment by bwbarrett** [comment]
Yes, this is expected.  We do not update the VERSION file to erase the "greek" value when tagging, as that would be an extra (useless) commit in our tree and makes some of our positive sanity checks (such as building the tarball from the same git hash for nightly testing and actual release) difficult.  The actual value of the greek field at a tag depends on how many times we messed up and had to create new release candidates.  Sometimes we did well and it's 'rc1' and sometimes we've done really poorly and it's 'rc7'.

We use the `contrib/dist/make_dist_tarball` script in the repository to make the actual release tarball, if you're looking to make a new release tarball.  But for the v3.0 and later releases, the release tarballs are always built using that script from the release tag itself, if that's what you are asking.
